File: src/main/java/ru/mdashlw/hypixel/reply/impl/PlayerReply.java
package ru.mdashlw.hypixel.reply.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import ru.mdashlw.hypixel.entities.player.HypixelPlayer;
import ru.mdashlw.hypixel.reply.Reply;

public final class PlayerReply extends Reply<HypixelPlayer> {
    @JsonProperty
    private JsonNode player;

    @Override
    public HypixelPlayer getValue() {
        return this.player == null || !this.player.isObject() ? null : new HypixelPlayer((ObjectNode) this.player);
    }
}

File: src/main/kotlin/ru/mdashlw/hypixel/reply/impl/PlayerReply.kt
package ru.mdashlw.hypixel.reply.impl

import ru.mdashlw.hypixel.elements.player.HypixelPlayer
import ru.mdashlw.hypixel.reply.Reply

class PlayerReply(
    success: Boolean,
    cause: String?,
    throttle: Boolean,
    player: HypixelPlayer?
) : Reply<HypixelPlayer>(success, cause, throttle, player)
